test/modules/jwt-encryption-service/JwtEncryptionService.test.ts

```diff
@@ -0,0 +1,29 @@
+import * as assert from 'assert';
+
+import JwtEncryptionService from 'app/modules/jwt-encryption-service/JwtEncryptionService';
+
+
+describe('JwtEncryptionService', () => {
+
+    const service: JwtEncryptionService = new JwtEncryptionService({
+        oath_scope_prefix: '',
+        chsUrl: '',
+        accountClientId: '',
+        accountRequestKey: 'pXf+qkU6P6SAoY2lKW0FtKMS4PylaNA3pY2sUQxNFDk=',
+        accountUrl: ''
+    });
+
+    it('should generate random nonce value in base64', () => {
+        const nonce = service.generateNonce();
+        const test = /[A-Za-z0-9+/=]/.test(nonce);
+        assert.equal(nonce[nonce.length - 1], '=');
+        assert.equal(test, true);
+    });
+
+    it('should create an encrypted state string using nonce value', async () => {
+        const nonce = '2dsa=';
+        const state = await service.jweEncodeWithNonce('http://example.com', nonce);
+        const test = /eyJhbGciOiJkaXIiLCJlbmMiOiJBMTI4Q0JDLUhTMjU2Iiwia2lkIjoia2V5In0..*/.test(state);
```

Comment:
What `eyJhbGciOiJkaXIiLCJlbmMiOiJBMTI4Q0JDLUhTMjU2Iiwia2lkIjoia2V5In0` is? Would't it better to decrypt it back and compare decrypted value?

Reply:
That would involve implementing a decode functionality for a test. What do you think?

Reply:
I do not see anything wrong in implementing decrypt function for the test especially that most of that heavy lifting is delegated to libraries. Feel better than the test that has magic number like `eyJhbGciO...`

These tests here look more like tests that needed to be added for sake of having them. Tests should be useful not just exist.